### PR TITLE
Eliminate the next portion of warnings

### PR DESCRIPTION
**Pull request by suricactus**
Well, there are zero warnings when using QField.

### CODE REVIEW

**Comment by nirvn** [comment]
Nice! Could we add a github PR test that would fail if any warning is found while compiling / launching QField? We don't want to slowly drift back to dozens of warnings we have to ignore.

**Comment by suricactus** [comment]
I don't really see a way, since most of the warnings are happening runtime in QML. I am not sure, but I think @m-kuhn have experimented with static code analyzer a few months ago. Just every few months someone needs to use "Advanced Bee Project" and watch for warnings.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 881b6c2b19b8089ddc820e6e050a488cb004af9b:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1467-30f62e8d746f77b4bf73764bd72017ca4fc58303-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1467-30f62e8d746f77b4bf73764bd72017ca4fc58303-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1467-30f62e8d746f77b4bf73764bd72017ca4fc58303-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1467-30f62e8d746f77b4bf73764bd72017ca4fc58303-x86.apk)

**Comment by m-kuhn** [comment]
Nice !
What we could do is to connect to QQmlEngine::warnings when running tests and starting a test app.

**Comment by suricactus** [comment]
Would not really help for now, a lot of the errors are caused runtime, but all we have are unit tests. Some warnings happened in very specific conditions after moving a slider within a relation reference, after closing a popup etc.
 
Have you tried this project? https://github.com/faaxm/spix 
We can write a typical user workflow and then listening for `QQmlEngine::warnings`?

**Comment by nirvn** [comment]
@suricactus , I still think there's value in _at least_ insuring that there are no warning on startup. Until very recently, we had a bunch.

**Comment by m-kuhn** [comment]
- just running the app and checking for warning is a rather simple smoke test
- we should add more unit tests for individual qml components, they can check that they behave correctly and they can check for warnings
- spix looks interesting, let's have a look at that early next year

File: src/qml/FeatureForm.qml [hunk 1]
```diff
@@ -591,6 +592,8 @@ Page {
       color: !model.constraintsHardValid ?  Theme.errorColor : !model.constraintsSoftValid ? Theme.warningColor : Theme.mainColor
     }
 
+    Material.foreground: undefined
```
Comment (m-kuhn) [inline src/qml/FeatureForm.qml:595]: What is this for exactly?

Comment (suricactus) [reply src/qml/FeatureForm.qml:595]: The `ToolBox` enters a binding loop implicitly: 

```QML PageHeader (parent or ancestor of Material): Binding loop detected for property "foreground"```

Comment (suricactus) [reply src/qml/FeatureForm.qml:595]: So I should set the `Material.foreground` either to `undefined` or color of our choise (while). I think undefined is the safer option.

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**